File: gettls/짝지어 제거하기.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int solution(string s)
{
    int answer = 0;
    stack<char> ss;
    ss.push(s[0]);

    for (int i = 1; i < s.size(); i++) {
        if (!ss.empty() && s[i] == ss.top()) ss.pop();
        else ss.push(s[i]);
    }

    if (ss.empty()) answer = 1;
    return answer;
}

File: WonseokLee/21.12.24 ~ 21.12.31/짝지어 제거하기.py
s = "baabaa"


def solution(s):
    get = []
    word = s

    for w in word:
        # 최초의 경우에는 스택에 추가
        if len(get) == 0:
            get.append(w)

        # 가장 마지막에 추가된 스택의 문자와, 다음 문자가 같다면
        elif get[-1] == w:
            get.pop()
        else:
            get.append(w)

    if len(get) == 0:
        return 1
    else:
        return 0

print(solution(s))